compyle/profile.py

```diff
@@ -0,0 +1,100 @@
+""" Utils for profiling kernels
+"""
+
+import time
+from contextlib import contextmanager
+from collections import defaultdict
+from .config import get_config
+
+
+_profile_info = defaultdict(lambda: {'calls': 0, 'time': 0})
+
+
+def record_profile(name, time):
+    global _profile_info
+    _profile_info[name]['time'] += time
+    _profile_info[name]['calls'] += 1
+
+
+@contextmanager
+def profile(name):
```

Comment:
This is the only problem, what do we call this?  Perhaps `profile` can be an object which has an `__enter__/__exit__` and a `__call__` so it can work as either a decorator or a context manager?  Otherwise, if you expect this to be used less, perhaps `profile_ctx`?